### ISSUE TITLE
How to disable the automatic upgrade of devtools?

### ISSUE THREAD

**Issue by evansherry** [flutter/devtools#3560]
After devtools is automatically upgraded to 2.9.2, when I run flutter attach, he will report the following error 
```
Unhandled exception:
Invalid argument(s): A directory corresponding to fileSystemPath "/Users/xx/.pub-cache/hosted/pub.flutter-io.cn/devtools-2.9.2/build" could not be found
#0 createStaticHandler (package:shelf_static/src/static_handler.dart:50:5)
#1 defaultHandler (package:devtools_server/src/external_handlers.dart:38:23)
<asynchronous suspension>
#2 serveDevTools (package:devtools_server/src/server.dart:215:15)
<asynchronous suspension>
#3 serveDevToolsWithArgs (package:devtools_server/src/server.dart:73:10)
<asynchronous suspension>
```

Use `dart pub global activate devtools -v 2.8.0 `to lower the devtools version, but it always automatically upgrades to 2.9.2, which makes me very distressed. 
How can I disable the automatic upgrade of devtools?

**Comment by evansherry** [comment]
The devtools was upgraded automatically just now, and I had to lower the version again.
Can I understand why the build file is missing in devtools 2.9.2?

**Comment by DanTup** [comment]
The update is likely being done by the editor. The next version of the VS Code extension has an option to disable this, detailed here:

https://github.com/flutter/devtools/issues/3549#issuecomment-1003026698

You can switch to the pre-release channel if you want the fix prior to the stable release.

Note: In the latest stable SDKs (and future SDKs), the DevTools version comes from the SDK and that setting won't do anything. It's only for older SDKs that do not ship with DevTools (and instead if must be used from Pub).

**Comment by kenzieschmoll** [comment]
The 2.9.2 release on pub has been retracted. please run `dart pub global activate devtools -v 2.8.0` to resolve this issue.

**Comment by DanTup** [comment]
@kenzieschmoll it's worth noting that retraction is only honoured by the latest SDKs (see https://github.com/flutter/devtools/issues/3549#issuecomment-1003962841), so anyone that hasn't upgraded yet may be "upgraded" again to 2.9.2 by the IDEs.

The latest version of the VS Code extension has a setting to disable automatic DevTools upgrades for anyone that's unable to upgrade yet (although the setting will become redundant as it only applies to older SDKs too, since DevTools is used from the SDK in the latest versions).

**Comment by kenzieschmoll** [comment]
I just published DevTools 2.9.3 to resolve this issue. VS code can continue to safely automatically upgrade.

**Comment by DanTup** [comment]
@kenzieschmoll thanks!
